src/Sarif/Readers/AlgorithmKindConverter.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+
+using Microsoft.CodeAnalysis.Sarif.Sdk;
+
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Readers
+{
+    public class AlgorithmKindConverter : JsonConverter
+    {
+        public static readonly AlgorithmKindConverter Instance = new AlgorithmKindConverter();
+
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(AlgorithmKind);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            switch ((string)reader.Value)
+            {
+                case "BLAKE-256": return AlgorithmKind.Blake256;
+                case "BLAKE-512": return AlgorithmKind.Blake512;
+                case "ECOH": return AlgorithmKind.Ecoh;
+                case "FSB": return AlgorithmKind.Fsb;
+                case "GOST": return AlgorithmKind.Gost;
+                case "Groestl": return AlgorithmKind.Groestl;
+                case "HAS-160": return AlgorithmKind.Has160;
+                case "HAVAL": return AlgorithmKind.Haval;
+                case "JH": return AlgorithmKind.JH;
+                case "MD2": return AlgorithmKind.MD2;
+                case "MD4": return AlgorithmKind.MD4;
+                case "MD5": return AlgorithmKind.MD5;
+                case "MD6": return AlgorithmKind.MD6;
+                case "RadioGatun": return AlgorithmKind.RadioGatun;
+                case "RIPEMD": return AlgorithmKind.RipeMD;
+                case "RIPEMD-128": return AlgorithmKind.RipeMD128;
+                case "RIPEMD-160": return AlgorithmKind.RipeMD160;
+                case "RIPEMD-320": return AlgorithmKind.RipeMD320;
+                case "SHA-1": return AlgorithmKind.Sha1;
+                case "SHA-224": return AlgorithmKind.Sha224;
+                case "SHA-256": return AlgorithmKind.Sha256;
+                case "SHA-384": return AlgorithmKind.Sha384;
+                case "SHA-512": return AlgorithmKind.Sha512;
+                case "SHA-3": return AlgorithmKind.Sha3;
+                case "Skein": return AlgorithmKind.Skein;
+                case "Snefru": return AlgorithmKind.Snefru;
+                case "Spectral Hash": return AlgorithmKind.SpectralHash;
+                case "SWIFFT": return AlgorithmKind.Swifft;
+                case "Tiger": return AlgorithmKind.Tiger;
+                case "Whirlpool": return AlgorithmKind.Whirlpool;
+            }
+
+            return AlgorithmKind.Unknown;
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            switch ((AlgorithmKind)value)
+            {
+                case AlgorithmKind.Blake256: { writer.WriteRawValue("\"\"BLAKE-256\"\""); return; }
```

Comment:
Good catch. Looks like this would provoke an actual issue at runtime. I will fix this and introduce a serialization test that exercises all these values.